src/libraries/Microsoft.PowerFx.Core/Functions/UserDefinedFunction.cs

```diff
@@ -222,6 +222,12 @@ public static IEnumerable<UserDefinedFunction> CreateFunctions(IEnumerable<UDF>
                 {
                     errors.Add(new TexlError(udf.Ident, DocumentErrorSeverity.Severe, TexlStrings.ErrUDF_FunctionNameRestricted, udfName));
                     continue;
+                }
+
+                if (udf.Args.Count > 30)

```

Comment:
nit - can we place this as a constant field?   
private const int MaxParameterCount = 30;